Sources/Basics/SQLiteBackedCache.swift

```diff
@@ -0,0 +1,246 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+
+import TSCBasic
+import TSCUtility
+
+/// SQLite backed persistent cache.
+public final class SQLiteBackedCache<Value: Codable>: Closable {
+    public typealias Key = String
+
+    public let name: String
+    public let fileSystem: TSCBasic.FileSystem
+    public let location: SQLite.Location
+    public let configuration: SQLiteBackedCacheConfiguration
+
+    private var state = State.idle
+    private let stateLock = Lock()
+
+    private let diagnosticsEngine: DiagnosticsEngine?
+    private let jsonEncoder: JSONEncoder
+    private let jsonDecoder: JSONDecoder
+
+    public init(name: String, location: SQLite.Location, configuration: SQLiteBackedCacheConfiguration = .init(), diagnosticsEngine: DiagnosticsEngine? = nil) {
```

Comment:
Add `name` to generalize usage

Reply:
maybe `tableName` would be more clear that this is used as such??

Reply:
also may need to validate it to make sure no spaces etc

Reply:
Added a code comment that says `tableName` should follow SQLite naming rulings.